File: analysis/analysers/parsers/DependencyParser/src/test/resources/analysis/contract/examples/cpp/src/de/sots/cellarsandcentaurs/domain/model/Creature.cpp
#include "Creature.h"
#include "../../application/CreatureFacade.h" // Für MONSTROSITY

namespace de::sots::cellarsandcentaurs::domain::model {

Creature::Creature(CreatureId id)
    : id_{std::move(id)}, type_{application::CreatureFacade::STANDARD_CREATURE_TYPE} {}

Creature::Creature(CreatureId id, CreatureType type)
    : id_{std::move(id)}, type_{type} {}

const CreatureId& Creature::getId() const noexcept { return id_; }
void Creature::setId(const CreatureId& id) { id_ = id; }

std::optional<CreatureType> Creature::getType() const { return type_; }
void Creature::setType(CreatureType type) { type_ = type; }

std::optional<ArmorClass> Creature::getArmorClass() const { return armorClass_; }
void Creature::setArmorClass(const ArmorClass& ac) { armorClass_ = ac; }

const std::unordered_map<SpeedType, Speed>& Creature::getSpeeds() const noexcept { return speeds_; }
void Creature::setSpeeds(const std::unordered_map<SpeedType, Speed>& speeds) { speeds_ = speeds; }

std::optional<HitPoints> Creature::getHitPoints() const { return hitPoints_; }
void Creature::setHitPoints(const HitPoints& hp) { hitPoints_ = hp; }

}

File: analysis/analysers/parsers/DependencyParser/src/test/resources/analysis/contract/examples/csharp/src/de/sots/cellarsandcentaurs/domain/model/Creature.cs
using de.sots.cellarsandcentaurs.application;

namespace de.sots.cellarsandcentaurs.domain.model;

[Fightable]
public class Creature
{
    private CreatureId id;
    private string name;
    private CreatureType type;
    private HitPoints hitPoints;
    private ArmorClass armorClass;
    private Map<SpeedType, Speed> speeds;

    public Creature(CreatureId id,string name, CreatureType type, HitPoints hitPoints, ArmorClass armorClass, Map<SpeedType, Speed> speeds)
    {
        this.name = name;
        this.type = CreatureFacade.STANDARD_CREATURE_TYPE;
        this.hitPoints = hitPoints;
        this.armorClass = armorClass;
        this.speeds = speeds;
        this.id = id;
    }

    public string GetName()
    {
        return name;
    }

    public void SetName(string name)
    {
        this.name = name;
    }

    public CreatureType GetType()
    {
        return type;
    }

    public void SetType(CreatureType type)
    {
        this.type = type;
    }

    public HitPoints GetHitPoints()
    {
        return hitPoints;
    }

    public void SetHitPoints(HitPoints hitPoints)
    {
        this.hitPoints = hitPoints;
    }

    public ArmorClass GetArmorClass()
    {
        return armorClass;
    }

    public void SetArmorClass(ArmorClass armorClass)
    {
        this.armorClass = armorClass;
    }

    public Speed GetSpeed()
    {
        return speed;
    }

    public void SetSpeed(Speed speed)
    {
        this.speed = speed;
    }
}
